### PR TITLE
Adding stcal-alignment sub-package.

### PR DESCRIPTION
**Pull request by mairanteodoro**
This PR addresses the implementation of the common-code sub-package `alignment`.

**Note**
The downstream job for romancal will succeed once [PR#919](https://github.com/spacetelescope/romancal/pull/919) has been approved.

**Checklist**
- [X] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [X] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [X] added relevant label(s)

### CODE REVIEW

File: src/stcal/alignment/resample_utils.py
Comment (kmacdonald-stsci) [inline src/stcal/alignment/resample_utils.py:1]: Is this file necessary?

**Review by kmacdonald-stsci — APPROVED** [review]

File: src/stcal/alignment/resample_utils.py
Comment (mairanteodoro) [reply src/stcal/alignment/resample_utils.py:1]: Not anymore. Thanks for catching that.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,11 +5,13 @@ on:
     branches:
       - main
       - '*x'
+      - stcal-alignment
     tags:
       - '*'
   pull_request:
     branches:
       - main
+      - stcal-alignment
```
Comment (nden) [inline .github/workflows/ci.yml:14]: This and line 8 will need to be removed before merging.

File: docs/conf.py [hunk 2]
```diff
@@ -38,19 +39,41 @@ def setup(app):
...
 
+# Configuration for intersphinx: refer to the Python standard library.
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3/", None),
+    "numpy": ("https://numpy.org/devdocs", None),
+    "scipy": ("http://scipy.github.io/devdocs", None),
+    "matplotlib": ("https://matplotlib.org/stable", None),
+    "gwcs": ("https://gwcs.readthedocs.io/en/latest/", None),
+    "astropy": ("https://docs.astropy.org/en/stable/", None),
+    "stdatamodels": ("https://stdatamodels.readthedocs.io/en/latest/", None),
```
Comment (nden) [inline docs/conf.py:50]: There shouldn't be any dependence on stdatamodels. Is there?

**Review by nden — CHANGES_REQUESTED** [review]
This needs a rebase - some of the changes in `CHANGES.rst` do not appear in the version on master. Also I left a couple internal comments.

File: docs/conf.py (hunk 2 above)
Comment (mairanteodoro) [reply docs/conf.py]: I'm not sure why that was there, but it was removed.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mairanteodoro) [reply .github/workflows/ci.yml]: Removed.

File: CHANGES.rst [hunk 3]
```diff
@@ -19,6 +19,8 @@ Other
 1.4.4 (2023-09-15)
 ==================
 
+- Added ``alignment`` sub-package. [#179]
```
Comment (nden) [inline CHANGES.rst:22]: This needs to be moved to the unreleased section.

**Review by nden — CHANGES_REQUESTED** [review]
This still does not look quite right as there are changes on main that are duplicated here. Can you do a rebase and move the change log?

File: CHANGES.rst (hunk 3 above)
Comment (mairanteodoro) [reply CHANGES.rst]: Now it's done.

**Review by nden — APPROVED** [review]
LGTM
